frontend/www/js/omegaup/arena/ranking.ts

```diff
@@ -25,127 +25,160 @@ interface Problem {
   visibility: number;
 }
 
+export interface RankingOptions {
+  startTime: Date | null;
+  finishTime: Date | null;
+  placesToShowInChart?: number;
+}
+
 export interface RankingRequest {
   problemsetId: number;
   scoreboardToken: string;
   currentUsername: string;
   navbarProblems: types.NavbarProblemsetProblem[];
 }
 
-// Implement this function in a new PR
-export function onRankingEvents(events: types.ScoreboardEvent[]): void {
-  // Don't trust input data (data might not be sorted)
-  // TODO: use events
-  events.sort((a, b) => a.delta - b.delta);
-  createChart();
-}
-
-export function createChart(): void {
-  // Implement this function in a new PR
-}
-
-export function updateProblemScore({
-  alias,
-  previousScore,
-  scoreboard,
-  username,
-}: {
-  alias: string;
-  previousScore: number;
-  scoreboard: types.Scoreboard;
-  username: string;
-}): types.ScoreboardRankingEntry[] {
-  return scoreboard.ranking.map((rank) => {
-    const ranking = rank;
-    if (ranking.username === username) {
-      ranking.problems = rank.problems.map((problem) => {
-        const problemRanking = problem;
-        if (problemRanking.alias == alias) {
-          const maxScore = getMaxScore(
-            myRunsStore.state.runs,
-            problemRanking.alias,
-            previousScore,
-          );
-          problemRanking.points = maxScore;
-        }
-        return problemRanking;
-      });
-      ranking.total.points = rank.problems.reduce(
-        (accumulator, problem) => accumulator + problem.points,
-        0,
-      );
-    }
-    return ranking;
-  });
-}
+export class Ranking {
+  private readonly startTime: Date | null;
+  private readonly finishTime: Date | null;
```

Comment:
```suggestion
  private readonly startTime: Date;
  private readonly finishTime: Date;
```

no tiene mucho sentido que estos sean nulos si nada más se usan para concursos, que no tienen estos límites nulos. si el objetivo es poder construir objetos de esta clase para mandar llamar a un método, mejor hay que evitar la construcción de esos objetos creando sólo uno y guardándolo _o_ regresando las funciones que se mandan llamar en todos lados a funciones sueltas.

Reply:
Ok, regreso todo a función como estaba antes,